Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/UnilateralLagrangianConstraint.h

```diff
@@ -34,124 +35,41 @@
 namespace sofa::component::constraint::lagrangian::model
 {
 
-template<class DataTypes>
-class UnilateralLagrangianConstraint : public core::behavior::PairInteractionConstraint<DataTypes>
+struct UnilateralLagrangianContactParameters : public BaseContactParams
```

Comment:
```suggestion
struct UnilateralLagrangianContactParameters final : public BaseContactParams
```
?

Reply:
It could be but in the meantime user might want to add other unilateral constraints enriching the parameter set, so a heritage might be a good solution.

Reply:
But it is a simple struct, so maybe you are tight, adding final might ease the readability of the api